test-utils/substrate-test-runner/src/utils.rs

```diff
@@ -0,0 +1,154 @@
+use crate::ChainInfo;
+use futures::{Sink, SinkExt};
+use sc_client_api::execution_extensions::ExecutionStrategies;
+use sc_executor::WasmExecutionMethod;
+use sc_informant::OutputFormat;
+use sc_network::{
+	config::{NetworkConfiguration, TransportConfig},
+	multiaddr,
+};
+use sc_service::config::KeystoreConfig;
+use sc_service::{BasePath, Configuration, DatabaseConfig, Role, TaskExecutor};
+use sp_keyring::Sr25519Keyring;
+use std::fmt;
+use std::io::Write;
+
+/// Used to create `Configuration` object for the node.
+pub fn config<Node>(task_executor: TaskExecutor) -> Configuration
+where
+	Node: ChainInfo,
+{
+	let mut chain_spec = Node::load_spec().expect("failed to load chain specification");
+	let base_path = if let Some(base) = Node::base_path() {
+		BasePath::new(base)
+	} else {
+		BasePath::new_temp_dir().expect("couldn't create a temp dir")
+	};
+	let root_path = base_path.path().to_path_buf().join("chains").join(chain_spec.id());
+
+	let role = Role::Authority {
+		sentry_nodes: Vec::new(),
+	};
+	let key_seed = Sr25519Keyring::Alice.to_seed();
+	let storage = chain_spec
+		.as_storage_builder()
+		.build_storage()
+		.expect("could not build storage");
+
+	chain_spec.set_storage(storage);
+
+	let mut network_config = NetworkConfiguration::new(
+		format!("Polkadot Test Node for: {}", key_seed),
```

Comment:
reading "Polkadot" again... why?

Is this is polkadot-specific, it should live there, not here.